helper.py

```diff
@@ -1,41 +1,113 @@
-#!/usr/bin/env python3
+'''
+IMPORTANT NOTE!!
 
-# This file may be obsolete - need to check if scaling in shephard is laggy
+=> The following will ONLY run on Python 3.7 to 3.9
+=> Install websockets version 11.0.3 via pip whl (do not use v12)
+=> Install asynio 0.2.0 via pip whl
+'''
 
-import os
+import asyncio
+import aionotify
+from websockets.server import serve
+import websockets
 import time
-
+import base64
+import os
+from io import BytesIO
 from PIL import Image
 
 
-static_path = 'shepherd/static/'
-input_file = static_path + "full_image.jpg"
-output_file = static_path + "image.jpg"
+CONNECTIONS = set()
+
+#The following sets up the asynchronous waiting for file change
+watcher = aionotify.Watcher()
+
+#TODO - Change the input_file to the location of the image the camera takes!
+static_path = '/home/pi/shepherd/shepherd/static/'
+input_file = static_path + "image.jpg"
+output_file = static_path + "resized_image.jpg"
 stat_file = static_path + "imgtime.txt"
 mod_time = 0
 out_width = 640.0
 out_height = 480.0
 
-while True:
-    time.sleep(0.1)
-    if os.path.exists(input_file):
-        stat_obj = os.stat(input_file)
-        if mod_time != stat_obj.st_mtime:
-            try:
-                img = Image.open(input_file)
-                width_i, height_i = img.size
-                width = width_i * 1.0
-                height = height_i * 1.0
-                scale_x = out_width / width
-                scale_y = out_height / height
-                if scale_x > scale_y:
-                    img = img.resize((int(width*scale_y), int(height*scale_y)), Image.ANTIALIAS)
-                else:
-                    img = img.resize((int(width*scale_x), int(height*scale_x)), Image.ANTIALIAS)
-
-                mod_time = stat_obj.st_mtime
-                img.save(output_file,"JPEG")
-                with open(stat_file, "w") as myfile:
-                    myfile.write(str(mod_time))
-            except IOError:
-                pass
+watcher.watch(alias='image', path=input_file, flags=aionotify.Flags.MODIFY)#sets up watcher
+
+def shrink_image():
+    for c in range(3):
+        time.sleep(0.1)#give it time to write the file.
+         try:#this runs until the bot has finished writing the image
+            img = Image.open(input_file)
+            img.load()
+            print("Opened successfully")
+            break
+        except:
+            print("Error opening file: attempt #"+str(c))
+
+    if c > 3:
+        continue#error with this file, go back and wait for next change.
```

Comment:
This could just be an else, which will only run if we go the whole way through the loop.
```suggestion
    else:
        continue  # Error with this file, go back and wait for next change.
```